tools/doprofile.py

```diff
@@ -216,11 +216,12 @@ def calculatePolygonProfile(self, geometry, crs, model, library):
                     offset = 0.0
                 srcArray = rasterBand.ReadAsArray(*srcOffset)
                 srcArray = srcArray*scale+offset 
-                srcArray = np.nan_to_num(srcArray)
                 masked = np.ma.MaskedArray(srcArray,
-                         mask=np.logical_or(srcArray == noData,
-                         np.logical_not(rasterizedArray)))
-                
+                            mask=np.logical_or.reduce((
```

Comment:
Just as a suggestion, I would not nest so many commands into each other. Try to create the mask for the masked array as a separate variable before. And instead of `np.logical_and` etc, use the `&` and `|` operators.